packages/react/slider/src/Slider.tsx

```diff
@@ -493,12 +497,18 @@ type SliderThumbPrimitive = Polymorphic.ForwardRefComponent<
 >;
 
 const SliderThumb = React.forwardRef((props, forwardedRef) => {
-  const { ref: collectionRef, index } = useSliderCollectionItem();
-  const ref = useComposedRefs(forwardedRef, collectionRef);
   const context = useSliderContext(THUMB_NAME);
+  const { getItems } = useCollectionItem();
+  const [thumb, setThumb] = React.useState<React.ElementRef<typeof SliderThumbImpl> | null>(null);
+  const composedRefs = useComposedRefs(forwardedRef, (node) => setThumb(node));
+  const index = React.useMemo(
+    () => (thumb ? getItems().findIndex((item) => item.ref.current === thumb) : 0),
+    [getItems, thumb]
+  );
```

Comment:
I thought this part would mean all thumbs would render at `0` on the range since we require a re-render from state to `findIndex` but seems okay. There's something about React that I still need to learn here apparently.

Reply:
I think the server is rendering all thumb index's as `0` right now but it appears to work fine here because we only have one thumb. `context.values[0]` will be accurate for the first but with more I think we'll see the mismatch still?

Reply:
Ah man you're right, my brain isn't working. I was expecting it to be positioned at `0` on the range but it is `0` value index haha. So yeah, issue still remains.

Reply:
I wonder then, if we want to tackle the SSR separately perhaps we choose not to render the thumbs while we're waiting on the node? it would at least appear more visually consistent until we can come up with an alternative counting method that works with `strict`?

Reply:
I can't get the `node` if I don't render the thumb... haha

Reply:
Ooo, I can `hidden` it tho... :+1:

Reply:
> I can't get the `node` if I don't render the thumb, the node is the thumb... 🙈 haha

haha, of course 😂

Reply:
@andy-hook I've hidden them for a render instead 🙂